image/image.go

```diff
@@ -48,18 +67,78 @@ func NewManifest(reader *tar.Reader, header *tar.Header) ImageManifest {
 	return manifest[0]
 }
 
+func NewImageConfig(reader *tar.Reader, header *tar.Header) ImageConfig {
+	size := header.Size
+	configBytes := make([]byte, size)
+	_, err := reader.Read(configBytes)
+	if err != nil && err != io.EOF {
+		panic(err)
+	}
+	var imageConfig ImageConfig
+	err = json.Unmarshal(configBytes, &imageConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	layerIdx := 0
+	for idx := range imageConfig.History {
+		if imageConfig.History[idx].EmptyLayer {
+			imageConfig.History[idx].ID = "<missing>"
+		} else {
+			imageConfig.History[idx].ID = imageConfig.RootFs.DiffIds[layerIdx]
+			layerIdx++
+		}
+	}
+
+	return imageConfig
+}
+
+func GetImageConfig(imageTarPath string, manifest ImageManifest) ImageConfig{
+	var config ImageConfig
+	// read through the image contents and build a tree
+	fmt.Println("Fetching image config...")
+	tarFile, err := os.Open(imageTarPath)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	defer tarFile.Close()
```

Comment:
Should the `defer tarFile.Close()` be immediately after `os.Open()`? Is there any situation where the code gets a file handle but then exits 1 before closing it?

Reply:
I'm not certain if there is a case where the file is open before exit is called, however, I still think this is the preferred way to go for a couple reasons:

1.  os.Exit cleans up file descriptors
2. since we're not guaranteed to have an open file we shouldn't attempt to close it until after we've checked it was opened with no error.

Reply:
That makes sense.